superagi/models/agent_execution.py

```diff
@@ -164,4 +164,61 @@ def assign_next_step_id(cls, session, agent_execution_id: int, next_step_id: int
         if next_step.action_type == "ITERATION_WORKFLOW":
             trigger_step = IterationWorkflow.fetch_trigger_step_id(session, next_step.action_reference_id)
             agent_execution.iteration_workflow_step_id = trigger_step.id
-        session.commit()
\ No newline at end of file
+        session.commit()
+
+    @classmethod
+    def get_execution_from_agent_id_with_status(cls,session,agent_id,status_filter):
+        db_agent_execution = session.query(AgentExecution).filter(AgentExecution.agent_id==agent_id,AgentExecution.status==status_filter).first()
+        return db_agent_execution
+    
+    @classmethod
+    def get_updated_execution_config_obj(cls,agent_execution):
+        agent_execution_configs={}
+        if agent_execution.goal is not None:
+            agent_execution_configs = {
+                "goal": agent_execution.goal,
+            }
+        
+        if agent_execution.instruction is not None:
+            agent_execution_configs["instructions"]=agent_execution.instruction,
+        return agent_execution_configs
+    
+    @classmethod
+    def get_all_executions_with_status_and_agent_id(cls,session,agent_id,ExecutionStateChangeConfigIn,current_status):
+        db_execution_arr=[]
+        if ExecutionStateChangeConfigIn.run_ids is not None:
+            db_execution_arr=session.query(AgentExecution).filter(AgentExecution.agent_id==agent_id,AgentExecution.status==current_status,AgentExecution.id.in_(ExecutionStateChangeConfigIn.run_ids)).all()
+        else:
+            db_execution_arr=session.query(AgentExecution).filter(AgentExecution.agent_id==agent_id,AgentExecution.status==current_status).all()
+        return db_execution_arr
+    
+    @classmethod
+    def get_all_executions_with_filter_config_and_agent_id(cls,session,agent_id,filter_config):
+        db_execution_arr=[]
+        if filter_config.run_ids is not None and filter_config.run_status_filter is not None and filter_config.run_status_filter in ["CREATED", "RUNNING", "PAUSED", "COMPLETED", "TERMINATED"]:
+            db_execution_arr=session.query(AgentExecution).filter(AgentExecution.agent_id==agent_id,AgentExecution.id.in_(filter_config.run_ids),AgentExecution.status==filter_config.run_status_filter).all()
```

Comment:
query = session.query(AgentExecution).filter(AgentExecution.agent_id==agent_id)

if condition1:
   query = query.filter(AgentExecution.id.in_(filter_config.run_ids))
   
if condition2:
   query = query.filter(AgentExecution.id.in_(filter_config.run_ids))
 
rename it find_executions_by_filter_config

Reply:
optimized and renamed